### DISCUSSION TITLE
Moving away of Gringotts?

### DISCUSSION THREAD

**Discussion by nikosgram** [nikosgram/gringotts#123] [Announcements]
Hello :) 

The last few years I'm fighting keeping Gringotts up to date with the latest versions of Minecraft. Although that is easy, what is not easy is the improvement of performance of the plugin as it is based on old technology that is not supported by neither spigot and paper. I was planning to move away of Ebean, but that's gonna cause a lot of compatibility issues.

https://github.com/nikosgram/gringotts/issues/108

I had a lot of ideas about what I could do, but I would like before moving forward with any of them to ask you.

Few ideas I had:

1. Creating a new plugin that replaces Gringotts, probably named something like Emeralds or Bezant. *Not sure about the name yet*
2. Remove backwards compatibility and focus on improving performance with the risk of plugin users losing their data. *Completely removing ebean support and moving into a new data system*
3. Create a memory cache overlay for the data system that will keep everything in the RAM and periodically push it into the SQLite file through ebean. *That will probably allow us to move away of ebean in the future*
4. Don't focus on improving performance, the plugin is working perfectly as it is.

What are your thoughts on this one? I would like to know your opinion, and if you have better ideas, please feel free to describe them in this discussion. <3

Thanks,
Nikolaos.

**Comment by spikehidden** [comment]
For myself I'm fine with the first three options.\
Even if losing all data would be annoying as I need to find a way for money exchange to the new currency then.

But it's still better as it is right now. The EssentialsX Plugin is warning about slow Economy responses on every start up since I use Gringotts and with Towny it is almost impossible to use as it already crashed my server a few times. (Did not happen without Gringotts)

For whatever you'll decide I'll thanks for updating Gringotts and maybe it's successor in the future!
